packages/core/expressions/src/components/RequestCard.vue

```diff
@@ -0,0 +1,252 @@
+    />
+    <div class="badges">
+      <KBadge
+        appearance="info"
+        class="protocol"
+      >
+        {{ protocol }}
+      </KBadge>
+      <KBadge
+        v-if="['http', 'https'].includes(protocol) && method"
+        :appearance="getMethodBadgeAppearance(method)"
+      >
+        {{ method?.toUpperCase() }}
+      </KBadge>
+    </div>
+    <div class="url">
+      {{ url }}
+    </div>
+    <div
+      v-if="sni"
+      class="sni"
+    >
+      <div class="section-title">
+        SNI
+      </div>
+      <div class="sni-content">
+        {{ sni }}
+      </div>
+    </div>
+    <div
+      v-if="headers && Object.keys(headers).length > 0"
+      class="headers"
+    >
+      <div class="section-title">
+        Headers
+      </div>
+      <ul
+        v-for="(values, key) in headers"
+        :key="key"
+        class="header"
+      >
+        <li class="header-key">
+          {{ key }}
+          <ul>
+            <template v-if="Array.isArray(values)">
+              <li
+                v-for="value in values"
+                :key="value"
+                class="header-value"
+              >
+                {{ value }}
+              </li>
+            </template>
+            <li
+              v-else
+              :key="values"
+              class="header-value"
+            >
+              {{ values }}
+            </li>
+          </ul>
+        </li>
+      </ul>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { computed } from 'vue'
+import { CloseIcon } from '@kong/icons'
+import { BadgeMethodAppearances, type BadgeAppearance } from '@kong/kongponents'
+
+const props = defineProps<{
+  active?: boolean;
+  protocol: string;
+  host: string;
+  port?: number;
+  path?: string;
+  method?: string;
+  headers?: { [k: string]: string | string[] };
+  sni?: string;
+}>()
+
+const emit = defineEmits<{
+  remove: []
+}>()
+
+const url = computed(() => {
+  const port = typeof props.port === 'number' && ((props.protocol === 'http' && props.port !== 80) || (props.protocol === 'https' && props.port !== 443)) ? props.port : undefined
+  const hostPort = [props.host, ...port ? [port] : []].join(':')
+  const origin = `${props.protocol}://${hostPort}`
+  const path = props.path ?? '/'
+
+  return `${origin}${path}`
+})
+
+function getMethodBadgeAppearance(method: string) {
+  const lowerCasedMethod = method.toLowerCase()
+
+  return (Object.values(BadgeMethodAppearances).includes(lowerCasedMethod as any)
+    ? lowerCasedMethod
+    : 'custom') as BadgeAppearance
+}
+</script>
+
+<style lang="scss" scoped>
+.request-card {
+  align-items: flex-start;
+  background-color: $kui-color-background;
+  border-radius: $kui-border-radius-30;
+  display: flex;
+  flex-direction: column;
+  gap: $kui-space-40;
+  justify-content: flex-start;
+  min-width: 0;
+  padding: $kui-space-50;
+  position: relative;
+
+  &::before {
+    border: 1px solid $kui-color-border-neutral-weaker;
```

Comment:
Can we use design token here?

Reply:
I checked all styles in this commit, and used design tokens as more as possible.